src/strings-in-block/class-attributes.php

```diff
@@ -13,7 +13,7 @@ public function find( \WP_Block_Parser_Block $block ) {
 		$strings = [];
 		$attrs   = $this->getAttributes( $block );
 
-		if ( $attrs ) {
+		if ( $attrs && isset( $block->blockName ) ) {
```

Comment:
@strategio 

I have a couple of questions here.

1.

`$block` is an object of type `\WP_Block_Parser_Block` which has a field called `blockName`: how cannot be set? I find it quite impossible. Perhaps it can be an empty string?

2.

if you use `$attrs` only when `$block->blockName` is set (or probably only when is not falsy), perhaps you could move this check right into `$this->getAttributes( $block )` changing it from that:

```php
return ! empty( $block->attrs ) && is_array( $block->attrs ) ? $block->attrs : [];
```

To that:
```php
return ! empty( $block->attrs ) && is_array( $block->attrs ) && $block->blockName ? $block->attrs : [];
```

By the way, can you please avoid double negations like in `! empty( $block->attrs )`? You can just write `$block->attrs` to get the same result:

```php
return $block->attrs && is_array( $block->attrs ) && $block->blockName ? $block->attrs : [];
```

Moreover, `attrs` is also a field of `WP_Block_Parser_Block`. The expression can be further simplified as such:

```php
return is_array( $block->attrs ) && $block->blockName ? $block->attrs : [];
```

Reply:
Right @andreasciamanna, I think I was induced into this with the existing code. Indeed `WP_Block_Parser_Block` has these properties defined.